triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -411,32 +400,18 @@ class Runner(
     Flow[TriggerMsg]
       .wireTap(logReceivedMsg _)
       .mapConcat[TriggerMsg]({
-        case CompletionMsg(c) =>
-          try {
-            commandIdMap.get(UUID.fromString(c.commandId)) match {
-              case None =>
-                List()
-              case Some(internalCommandId) =>
-                List(CompletionMsg(c.copy(commandId = internalCommandId)))
-            }
-          } catch {
-            // This happens for invalid UUIDs which we might get for
-            // completions not emitted by the trigger.
-            case _: IllegalArgumentException => List()
-          }
-        case TransactionMsg(t) =>
-          try {
-            commandIdMap.get(UUID.fromString(t.commandId)) match {
-              case None =>
-                List(TransactionMsg(t.copy(commandId = "")))
-              case Some(internalCommandId) =>
-                List(TransactionMsg(t.copy(commandId = internalCommandId)))
-            }
-          } catch {
-            // This happens for invalid UUIDs which we might get for
-            // transactions not emitted by the trigger.
-            case _: IllegalArgumentException => List(TransactionMsg(t.copy(commandId = "")))
-          }
+        case msg @ CompletionMsg(c) =>
+          // This happens for invalid UUIDs which we might get for
+          // completions not emitted by the trigger.
+          val uuid = fromTryCatchThrowable[UUID, IllegalArgumentException](
+            UUID.fromString(c.commandId)).toOption
+          (uuid exists commandIdsUsed).option(msg).toList
+        case msg @ TransactionMsg(t) =>
+          // This happens for invalid UUIDs which we might get for
+          // transactions not emitted by the trigger.
+          val uuid = fromTryCatchThrowable[UUID, IllegalArgumentException](
+            UUID.fromString(t.commandId)).toOption
+          List(if (uuid exists commandIdsUsed) msg else TransactionMsg(t.copy(commandId = "")))
```

Comment:
Looks like we should be able to pop from that set now if we’ve seen both the transaction as well as the completion for a command. Doesn’t have to be in this PR.

Reply:
#7598